lib/src/puppetdb-cli.cc

```diff
@@ -155,19 +159,45 @@ size_t write_body(char *ptr, size_t size, size_t nmemb, void *userdata){
     return size * nmemb;
 }
 
+string
+convert_query_to_post_data(const string& query_str) {
+    // If this is PQL then we need to wrap the query in double-quotes, otherwise
+    // the query is AST and we leave it alone
+    rapidjson::StringBuffer s;
+    rapidjson::Writer<rapidjson::StringBuffer> writer(s);
+    writer.StartObject();
+    writer.String("query");
+
+    string query_str_copy { query_str };
```

Comment:
[2/4] Why memory is hard in C

Here we're copying the contents of the string reference to another string. We're really looking here for whether or not we have an AST-like query or PQL-like query. We copy and trim the string in case the user has some leading spaces.